plugin/trino-hive/src/test/java/io/trino/plugin/hive/HiveHadoopQueryRunner.java

```diff
@@ -0,0 +1,111 @@
+import io.trino.spi.security.SelectedRole;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.tpch.TpchTable;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.airlift.testing.Closeables.closeAllSuppress;
+import static io.trino.plugin.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+import static io.trino.spi.security.SelectedRole.Type.ROLE;
+import static io.trino.testing.QueryAssertions.copyTpchTables;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+
+public final class HiveHadoopQueryRunner
+{
+    private static final SelectedRole ADMIN_ROLE = new SelectedRole(ROLE, Optional.of("admin"));
+
+    private HiveHadoopQueryRunner() {}
+
+    static {
+        System.setProperty("HADOOP_USER_NAME", "hive");
+    }
+
+    public static DistributedQueryRunner createHadoopQueryRunner(
+            HiveHadoop server,
+            Map<String, String> extraProperties,
+            Map<String, String> connectorProperties,
+            Iterable<TpchTable<?>> tables)
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(createSession())
+                .setExtraProperties(extraProperties)
+                .build();
+        try {
+            queryRunner.installPlugin(new TpchPlugin());
+            queryRunner.installPlugin(new TestingHivePlugin());
+            queryRunner.createCatalog("tpch", "tpch");
+
+            connectorProperties = new HashMap<>(ImmutableMap.copyOf(connectorProperties));
+            connectorProperties.putIfAbsent("hive.metastore.uri", "thrift://" + server.getHiveMetastoreEndpoint().toString());
+            connectorProperties.putIfAbsent("hive.metastore.thrift.client.socks-proxy", server.getSocksProxyEndpoint().toString());
+            connectorProperties.putIfAbsent("hive.hdfs.socks-proxy", server.getSocksProxyEndpoint().toString());
+            connectorProperties.putIfAbsent("hive.max-partitions-per-scan", "1000");
+            connectorProperties.putIfAbsent("hive.metastore-timeout", "30s");
+            connectorProperties.putIfAbsent("hive.security", "sql-standard");
+            connectorProperties.putIfAbsent("hive.hive-views.enabled", "true");
+
+            queryRunner.createCatalog("hive", "hive", connectorProperties);
+
+            queryRunner.execute(createSession(), "CREATE SCHEMA tpch");
+            copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, createSession(), tables);
+            return queryRunner;
+        }
+        catch (Throwable e) {
+            closeAllSuppress(e, queryRunner);
+            throw e;
+        }
+    }
+
+    private static Session createSession()
+    {
+        return testSessionBuilder()
+                .setIdentity(Identity.forUser("hive")
+                        .withConnectorRoles(ImmutableMap.of(
+                                "hive", ADMIN_ROLE,
+                                "tpch", ADMIN_ROLE))
```

Comment:
This looks needed because of sql-standard, but sql-standard shouldn't be the default, unless it's inherently required to work with a dockerized hive. is it?

Reply:
What security should be used? I used `sql-standard` since it's `HiveQueryRunner`'s default.

Reply:
HiveQueryRunner shouldn't have different defaults than hive connector itself .